src/services/external-api-helper.ts

```diff
@@ -0,0 +1,275 @@
+   * queue.
+   */
+  if (!imdbId) {
+    const parsedFilename = episodeParser(searchRequest.name);
+    const isTVEpisode = Boolean(parsedFilename && parsedFilename.show && parsedFilename.season && parsedFilename.episode);
+    if (isTVEpisode) {
+      searchRequest.name = parsedFilename.show;
+      searchRequest.reqtype = 'series';
+      const tvSeriesInfo = await imdbAPI.get(searchRequest);
+
+      if (tvSeriesInfo && tvSeriesInfo instanceof TVShow) {
+        const allEpisodes = await tvSeriesInfo.episodes();
+        const currentEpisode = _.find(allEpisodes, { season: parsedFilename.season, episode: parsedFilename.episode });
+        if (!currentEpisode) {
+          throw new IMDbIDNotFoundError();
+        }
+
+        imdbId = currentEpisode.imdbid;
+      }
+    }
+
+    if (!imdbId) {
+      searchRequest.reqtype = 'movie';
+      const searchResults = await imdbAPI.search(searchRequest);
+      // find the best search results utilising the Jaro-Winkler distance metric
+      const searchResultStringDistance = searchResults.results.map(result => natural.JaroWinklerDistance(searchRequest.name, result.title));
+      const bestSearchResultKey = _.indexOf(searchResultStringDistance, _.max(searchResultStringDistance));
+
+      const searchResult: any = searchResults.results[bestSearchResultKey];
+      if (!searchResult) {
+        throw new IMDbIDNotFoundError();
+      }
+
+      imdbId = searchResult.imdbid;
+    }
+  }
+
+  const imdbData = await imdbAPI.get({ id: imdbId });
+  if (!imdbData) {
+    return null;
+  }
+
+  let metadata;
+  if (imdbData.type === 'movie') {
+    metadata = mapper.parseIMDBAPIMovieResponse(imdbData);
+  } else if (imdbData.type === 'series') {
+    metadata = mapper.parseIMDBAPISeriesResponse(imdbData);
+  } else if (imdbData.type === 'episode') {
+    const tvSeriesId = (imdbData as Episode).seriesid;
+    const existingSeries: SeriesMetadataInterface = await SeriesMetadata.findOne({ imdbID: tvSeriesId }, null, { lean: true }).exec();
+    if (!existingSeries) {
+      try {
+        await EpisodeProcessing.create({ seriesimdbid: tvSeriesId });
+      } catch (e) {
+        if (e.code !== MONGODB_DUPLICATE_KEY_ERROR_CODE) {
+          throw e;
+        }
+      }
+    }
+    metadata = mapper.parseIMDBAPIEpisodeResponse(imdbData);
+  } else {
+    throw new Error('Received a type we did not expect');
+  }
+
+  return metadata;
+};
+
+/**
+ * Attempts a query to the IMDb API and standardizes the response
+ * before returning.
+ *
+ * IMDb does not return goofs, osdbHash, tagline, trivia
+ *
+ * @param [imdbId] the IMDb ID
+ * @param [searchRequest] a query to perform in order to get the imdbId
+ * @param [season] the season number if this is an episode
+ * @param [episode] the episode number if this is an episode
+ */
+export const getFromIMDbAPIV2 = async(imdbId?: string, searchRequest?: SearchRequest, season?: number, episode?: number): Promise<MediaMetadataInterface> => {
```

Comment:
Note, this method has changed a bit, so review should take that into account (i.e. it has been moved, but also changed)